### PR TITLE
drivers: gps: nrf9160_gps: Enable restart of already running gps.

### PR DESCRIPTION
**Pull request by jhn-nordic**
Now the gps will be restarted if already running when a start
request is sent.

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 1]
```diff
@@ -480,11 +485,6 @@ static int parse_cfg(struct gps_config *cfg_src,
 			return -EINVAL;
 		}
 
-		if (cfg_src->timeout >= cfg_src->interval) {
```
Comment (jtguggedal) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:483]: Might split this change into separate commit

**Review by jtguggedal — APPROVED** [review]

File: drivers/gps/nrf9160_gps/nrf9160_gps.c (hunk 1 above)
Comment (jhn-nordic) [reply drivers/gps/nrf9160_gps/nrf9160_gps.c:483]: done

**Review by simensrostad — APPROVED** [review]
LGMT

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 2]
```diff
@@ -197,6 +196,11 @@ static int open_socket(struct gps_drv_data *drv_data)
 	return 0;
 }
 
+static void cancel_works(struct gps_drv_data *drv_data)
```
Comment (jhn-nordic) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:199]: Will get more than one "k_delayed_work_cancel" in upcoming PR.

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 3]
```diff
@@ -522,8 +520,8 @@ static int start(struct device *dev, struct gps_config *cfg)
 	}
 
 	if (atomic_get(&drv_data->is_active)) {
-		LOG_WRN("GPS is already active");
-		return -EALREADY;
+		LOG_DBG("GPS is already active. Clean up before restart");
+		cancel_works(drv_data);
```
Comment (anangl) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:524]: It would be nice to mention it in the API description that `start_gps` will restart the GPS if it is already started. Especially that this changes the behavior of an existing API (even though the previous behavior was not strictly specified).

Comment (jhn-nordic) [reply drivers/gps/nrf9160_gps/nrf9160_gps.c:524]: Added comment in gps.h

File: include/drivers/gps.h [hunk 4]
```diff
@@ -196,6 +196,9 @@ typedef void (*gps_event_handler_t)(struct device *dev,
  * @typedef gps_start_t
  * @brief Callback API for starting GPS operation.
  *
+ * If gps is already running a call to this function will
+ * restart the gps.
+ *
```
Comment (anangl) [inline include/drivers/gps.h:201]: Please move it to the description of `gps_start()`. That's where users will look for information regarding the function behavior.

Comment (jhn-nordic) [reply include/drivers/gps.h]: done. That was actually where I intended it to be :)

**Review by anangl — APPROVED** [review]
